actors/evm/src/interpreter/system.rs

```diff
@@ -111,44 +111,70 @@ pub struct System<'r, RT: Runtime> {
     pub readonly: bool,
     /// Randomness taken from the current epoch of chain randomness
     randomness: OnceCell<[u8; 32]>,
+
+    /// This is "some" if the actor is currently a "zombie". I.e., it has selfdestructed, but the
+    /// current message is still executing.
+    tombstone: Option<Tombstone>,
 }
 
 impl<'r, RT: Runtime> System<'r, RT> {
-    /// Create the actor.
-    pub fn create(rt: &'r mut RT) -> Result<Self, ActorError>
+    fn new(rt: &'r mut RT, readonly: bool) -> Self
     where
         RT::Blockstore: Clone,
     {
-        let read_only = rt.read_only();
-        let state_root = rt.get_state_root()?;
-        if state_root != EMPTY_ARR_CID {
-            return Err(actor_error!(illegal_state, "can't create over an existing actor"));
-        }
         let store = rt.store().clone();
-        Ok(Self {
+        Self {
             rt,
             slots: StateKamt::new_with_config(store, KAMT_CONFIG.clone()),
             nonce: 1,
             saved_state_root: None,
             bytecode: None,
-            readonly: read_only,
+            readonly,
             randomness: OnceCell::new(),
-        })
+            tombstone: None,
+        }
+    }
+    /// Create the actor.
+    pub fn create(rt: &'r mut RT) -> Result<Self, ActorError>
+    where
+        RT::Blockstore: Clone,
+    {
+        let read_only = rt.read_only();
+        let state_root = rt.get_state_root()?;
+        if state_root != EMPTY_ARR_CID {
```

Comment:
Mild discomfort level due to https://github.com/filecoin-project/ref-fvm/issues/1359.

Reply:
These aren't quite the same thing. Here, we're not assuming this block _exists_, we're checking to see if we've been initialized. The system _does_ guarantee that the state-root starts out as the empty array CID (this specific logic is present in all actors).